File: robot_ws/src/robot_py/robot_py/suscriptor_simple.py
# Importaciones necesarias
import rclpy                    # Librería principal de ROS2 para Python
from rclpy.node import Node    # Clase base para crear nodos
from std_msgs.msg import String # Tipo de mensaje String

class SuscriptorSimple(Node):
    def __init__(self):
        # Llamamos al constructor de la clase padre (Node) 
        # con el nombre "suscriptor_simple"
        super().__init__("suscriptor_simple")
        
        # Creamos la suscripción con estos parámetros:
        # 1. String: tipo de mensaje que vamos a recibir
        # 2. "chatter": nombre del topic al que nos suscribimos
        # 3. self.msgCallback: función que se ejecutará cuando llegue un mensaje
        # 4. 10: tamaño de la cola de mensajes
        self.sub_ = self.create_subscription(String, 
                                           "chatter",
                                           self.msgCallback, 
                                           10)

    def msgCallback(self, msg):
        # Función que se ejecuta cada vez que llega un mensaje
        # msg: es el mensaje recibido, de tipo String
        # Imprimimos el contenido del mensaje usando el logger
        self.get_logger().info("He escuchado: %s" % msg.data)

def main():
    # Inicializamos ROS2
    rclpy.init()
    
    # Creamos una instancia de nuestro nodo suscriptor
    suscriptor_simple = SuscriptorSimple()
    
    # Mantenemos el nodo ejecutándose
    rclpy.spin(suscriptor_simple)
    
    # Limpieza al terminar
    suscriptor_simple.destroy_node()
    rclpy.shutdown()

# Este if asegura que el código solo se ejecute si el script
# se ejecuta directamente (no si se importa como módulo)
if __name__ == "__main__":
    main()

File: robot_ws/src/robot_cpp/src/suscriptor_simple.cpp
// Incluimos las cabeceras necesarias
#include <rclcpp/rclcpp.hpp>        // Librería principal de ROS2
#include <std_msgs/msg/string.hpp>   // Para usar mensajes de tipo String

using std::placeholders::_1;         // Para usar std::bind con un argumento

// Definimos una clase que hereda de rclcpp::Node
class SimpleSubscriber : public rclcpp::Node {
public:
    // Constructor de la clase
    SimpleSubscriber() : Node("simple_subscriber") {
        // Creamos la suscripción con estos parámetros:
        // 1. "chatter": nombre del topic
        // 2. 10: tamaño de la cola de mensajes
        // 3. std::bind: vincula la función callback con this (objeto actual) y _1 (el mensaje)
        sub_ = create_subscription<std_msgs::msg::String>(
            "chatter", 
            10, 
            std::bind(&SimpleSubscriber::msgCallback, this, _1)
        );
    }

private:
    // Declaramos el suscriptor como variable miembro privada
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;

    // Función callback que se ejecuta cuando llega un mensaje
    void msgCallback(const std_msgs::msg::String &msg) const {
        // Imprime el mensaje recibido usando el logger de ROS
        RCLCPP_INFO_STREAM(get_logger(), "I heard: " << msg.data.c_str());
    }
};

// Función principal
int main(int argc, char * argv[]) {
    rclcpp::init(argc, argv);    // Inicializamos ROS2
    
    // Creamos una instancia compartida de nuestro nodo
    auto node = std::make_shared<SimpleSubscriber>();
    
    // Mantenemos el nodo ejecutándose
    rclcpp::spin(node);
    
    // Cerramos ROS2 limpiamente
    rclcpp::shutdown();
    return 0;
}